pkg/controllers/sync/controller.go

```diff
@@ -550,7 +550,10 @@ func (s *SyncController) syncToClusters(
 		isSelectedCluster := selectedClusterNames.Has(clusterName)
 		isCascadingDeletionTriggered := cluster.GetDeletionTimestamp() != nil &&
 			cascadingdeletion.IsCascadingDeleteEnabled(cluster)
-		shouldBeDeleted := !isSelectedCluster || isCascadingDeletionTriggered
+		if isCascadingDeletionTriggered {
+			continue
```

Comment:
i'm not sure if it's sound to immediately skip processing a cluster at this point. here are some questions to consider:

1. if a cluster is not selected but previously contained the object, we now defer its deletion to the cascading-deletion process while the current sync reports a success. is this acceptable to the user? (sync success is reported even though it does not match the desired placement)
2. if a cluster is not selected but previously contained the object (same scenario as 1), what is the new FederatedObject status after the sync is complete? should it (and does it) reflect the status of the skipped cluster in any way? 
3. after an object is deleted by the cascading-delete loop, is the corresponding FederatedObject's status updated to reflect the new state?

Reply:
cc @gary-lgy @mrlihanbo

Reply:
The latest code adds a FederatedObject status update operation in the `syncToClusters` function: 
- If cascading deletion is triggered, wait for `reconcileClusterForCascadingDeletion` to perform the deletion operation. 
- If the resource has not been deleted yet, update the FederatedObject status to wait for deletion.

```go
if isCascadingDeletionTriggered {
	if clusterObj != nil && clusterObj.GetDeletionTimestamp() == nil {
		dispatcher.RecordStatus(clusterName, fedcorev1a1.WaitingForRemoval)
	}
	continue
}
```

Reply:
This resolves the first 2 issues I have, but what about after the object is deleted in the cascading-delete loop? Is the corresponding FederatedObject's status updated to remove the cluster?

Reply:
When the member cluster is completely deleted, the FederatedObject object will be re-queued for the sync process. Since the member cluster was deleted, it won't be included in this synchronization status set. 
(Since FederatedObject's status is fully updated every time, the cluster status not included in the status set will be deleted.)